docs/adr/adr-004-state-interaction.md

```diff
@@ -0,0 +1,173 @@
+# ADR #004: Celestia-Node State Interaction for March 2022 Testnet
+
+<hr style="border:3px solid gray"> </hr>
+
+## Authors
+
+@renaynay
+
+## Changelog
+
+* 2022-01-14: initial draft
+
+<hr style="border:2px solid gray"> </hr>
+
+## Context
+
+Currently, celestia-node lacks a way for users to submit a transaction to celestia-core.
+
+## Decision
+
+Both celestia **light** and **full** nodes will run `StateService`. 
+`StateService` will be responsible for lifecycle of the given implementation of `StateAccessor`.
+
+`StateAccessor` will be an interface for interacting with celestia-core in order to retrieve account-related information
+as well as submit transactions.
+
+### `StateService`
+
+`StateService` will manage the lifecycle of the `StateAccessor` and also expose several higher-level RPC endpoints for 
+users to access the methods provided by the `StateAccessor`.
+
+```go
+type StateService struct {
+    accessor StateAccessor
+}
+``` 
+
+### `StateAccessor`
+
+`StateAccessor` is defined by the two methods listed below and may be expanded in the future to include more methods 
+related to accounts and transactions in later iterations.
+
+```go
+type StateAccessor interface {
+    func AccountBalance(context.Context, Account) (Balance, error)
```

Comment:
We should still think through this. Accessing Celestia's balance is not enough and there are many more fields, like delegation that light clients need to access.

Reply:
Yes, later iterations please.

Reply:
resolved?

Reply:
Yes